Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -302,7 +302,7 @@ public final class ManifestLoader: ManifestLoaderProtocol {
                                                 swiftpmVersion: Versioning.currentVersion.displayString,
                                                 fileSystem: fileSystem)
 
-            if self.useInMemoryCache, let manifest = self.memoryCache[cacheKey] {
+            if self.useInMemoryCache, let manifest = self.memoryCache[cacheKey], manifest.url == baseURL {
```

Comment:
I think this is fine, we can also just remove the in-memory cache if we are worried that URL is not the only attribute that could potentially be changes (tho right now its the only issue afaict)

Reply:
That makes sense to me.  Would you prefer actually removing the code for the cache or disabling it (using compilation conditionals)?  I imagine that it would come back in some form, though perhaps it would be different enough that it would be worth starting with clean code.  I suppose it can always be brought back from Git history if needed, so maybe it's not really even worth considering keeping it in the code.

Reply:
I think the amount of code involved is minimal, so not worth keeping around. worth testing tho how much performance we loose in larger projects buy removing it

Reply:
Agreed.  I'll do a test using a large test package with many dependencies.